Jenkinsfile

```diff
@@ -5,17 +5,40 @@ elifeLibrary {
         commit = elifeGitRevision()
     }
 
-    stage 'Build image', {
-        sh 'docker build -t elife/sciencebeam-utils .'
-    }
+    node('containers-jenkins-plugin') {
+        def candidateVersion
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+            try {
+                candidateVersion = sh(script: "IMAGE_TAG=${commit} docker-compose run --rm ./print_version.sh", returnStdout: true).trim()
+                echo "Candidate version: v${candidateVersion}"
+            } finally {
+                sh 'docker-compose down -v'
+            }
+        }
+
+        stage 'Project tests', {
+            try {
+                sh "IMAGE_TAG=${commit} docker-compose run --rm sciencebeam-utils ./project_tests.sh"
+            } finally {
+                sh 'docker-compose down -v'
+            }
+        }
+
+        stage 'Push release', {
+            // def isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
+            // if (isNew) {
+            //     sh "git tag v${candidateVersion} && git push origin v${candidateVersion}"
+            //     sh "twine upload dist/*"
+            // }
+        }
 
-    stage 'Run tests', {
-        elifeLocalTests './project_tests.sh'
     }
 
     elifeMainlineOnly {
-        stage 'Merge to master', {
-            elifeGitMoveToBranch commit, 'master'
+        stage 'Approval', {
```

Comment:
not sure if we need this

Reply:
only if this project is deployed somewhere, but it seems a library so just `develop` and `master` should be good?

Reply:
Yes, no deployment. Can we also omit the 'Merge to master', it appears to be implicit (or at least I didn't see it in as much config anymore)?

Reply:
There is no `Jenkinsfile.prod` nor production deployment, so this should be
```
elifeGitMoveToBranch commit, `master`
```
so that `master` is always the latest tested code. No need for `approved`.

Reply:
(you don't usually see it because it's in another pipeline, `prod-$project`)